### ISSUE TITLE
CombinedChart CandleStick real body not showing

### ISSUE THREAD

**Issue by rayarteagas** [PhilJay/MPAndroidChart#4282]
**Summary**
With a CombinedChart, The CandleSticks real bodies are not displaying. Only the shadows are rendered.

**Expected Behavior**
I expect the real bodies to be rendered with the declared decreasing/neutral/increasing color.

**Possible Solution**
--

**Device (please complete the following information):**
 - Device: Xiaomi Redmi Note 4
 - Android Version: 7.0
 - Library Version 3.0.3

**Additional Context**
the data is coming from: https://min-api.cryptocompare.com/data/histoday?fsym=BTC&tsym=USD&aggregate=1&G&limit=10&e=CCCAGG
https://gist.github.com/rayarteagas/fa9242ac9a31db20782b2802d8e5010e

**Comment by almic** [comment]
I'm not sure how that's happened, and I would test it myself, but right now I can't exactly do that. I'm adding to the support list and hopefully you get an answer.

**Comment by rayarteagas** [comment]
Wen the X values are very long (like in a timestamp for example), the real body shrinks horizontally and, at some point, hides no matter how much you zoom. The solution I'm using right now is to scale the X values (for example dividing by 10000 or so), and then reverting the scaled values to the original ones in the value formatters and marker views. This does not happens with the shadows, so I think is a math problem.

**Comment by almic** [comment]
I don't believe that having long labels should be causing the boxes to shrink at all. I wonder if this happens to bar charts or is a unique problem to the candlestick chart. Definitely a problem!

**Comment by rayarteagas** [comment]
I'm using a CombinedChart with lines and candlesticks, but the problem is also on CandleScrickChart. I'll take some time to simplify the code posted, so the issue may be more clear.

**Comment by ArcherEmiya05** [comment]
I know this is old but @rayarteagas is correct, if your `Entry.x` value which is the timestamp for this library's chart is too long or too large e.g. `1705795200000` this will shrink the whole body. The temporary fix is to reduce that value by dividing it with e.g. `86400000f` which is equivalent to one day milliseconds. Just make sure to multiply it back with same value `86400000f` when showing it with `MarkerView.refreshContent`.
